web/src/components/apps/DashboardVersionCard.jsx

```diff
@@ -907,16 +909,28 @@ class DashboardVersionCard extends React.Component {
       return null;
     }
 
+    let latestVersion = downstream?.latestVersion;
+    let newVersionsAfterUpgrade = null;
+    let versionsToSkip = downstream?.pendingVersions?.length - 1;
+    const requiredVersions = downstream?.pendingVersions?.length > 0 ? filter(downstream.pendingVersions, ["isRequired", true]) : [];
+    if (this.props.currentVersion?.deployedAt && requiredVersions.length > 0) { // If there is a version already deployed, and there is at least one required pending versions, set latestVersion to the earliest required version
+      const sortedVersions = orderBy(requiredVersions, ["createdOn", "desc"]);
+      latestVersion = sortedVersions[0];
+      const indexOfLatestVersion = findIndex(downstream?.pendingVersions, ["versionLabel", latestVersion.versionLabel]);
+      versionsToSkip = (downstream?.pendingVersions?.length - 1) - indexOfLatestVersion;
```

Comment:
shouldn't this just be
```suggestion
      versionsToSkip = indexOfLatestVersion;
```

Reply:
I don't think so. `indexOfLatestVersion` is just finding it's position in the array so for example it might be a the second position in the array but there could be 6 versions between it and the one that's currently deployed so showing "2 versions will be skipped...." would be incorrect. 

This number has been correct in all the scenarios i've tested. I'm trying to get loom working so I can get a demo video recorded to be able to see it all.

Reply:
how would that even happen? pending versions is the list of versions above the currently deployed version, and the index of the "nextAppVersion" should be exactly the number of versions between the currently deployed and the next app version.